src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+                case 'us':
+                    defaultDateFormat = 'mm/dd/yy';
+                    inputPlaceholder = 'mm/dd/yyyy';
+                    break;
+                case 'reverse':
+                    defaultDateFormat = 'yy/mm/dd';
+                    inputPlaceholder = 'yyyy/mm/dd';
+                    break;
+                default:
+                    defaultDateFormat = 'dd/mm/yy';
+                    inputPlaceholder = 'dd/mm/yyyy'
+            }
+
+            // Initialize date pickers
+            $datePickerInput.datepicker({
+                closeText: 'Close',
+                dayNamesShort: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"],
+                dateFormat: defaultDateFormat,
+                onClose: this.utils.removeAria.bind(this, $html),
+                showAnim: '',
+                showButtonPanel: true,
+                showOn: 'button',
+                // Hide the default JQUI trigger button so we can attach to our own
+                beforeShow: function(input, inst){
+                    $(inst.dpDiv).addClass('pulsar-datepicker');
+                }
+            });
+
+            // Add placeholder matching the date format, allowing for difference in string format
+            $datePickerInput.attr('placeholder', inputPlaceholder);
+
+            // Switch off autocomplete to avoid it overlapping the date picker
+            $datePickerInput.attr('autocomplete', 'off');
+
+            // Attach enhancements to the linked trigger element
+            $linkedTriggerButton.on('click', this.enhanceDatePicker.bind(this, $datePickerInput));
+
+        });
+
+        // Attach close listener
+        $html.on('click', '#ui-datepicker-div .ui-datepicker-close', () => {
+            this.closeCalendar();
+        });
+    }
+
+    /**
+     * Enhance the jQuery UI datepicker on trigger button click event
+     * @param {Event} event
+     */
+    enhanceDatePicker ($datePickerInput, event) {
+        // Prevent default
+        event.preventDefault();
+
+        // Get the triggering button and related date input
+        this.$triggerButton = $(event.target);
+        //this.$dateInput = this.$triggerButton.closest('.controls').find('[data-datepicker]');
+        this.$dateInput = $datePickerInput;
+
+        // Show the datepicker
+        this.$dateInput.datepicker('show');
+
+        // Get the datepicker container
+        this.$datePickerContainer = this.$html.find('#ui-datepicker-div');
+
+        // Get the next/prev link controls
+        this.$prevLink = this.$datePickerContainer.find('.ui-datepicker-prev');
+        this.$nextLink = this.$datePickerContainer.find('.ui-datepicker-next');
+
+        // Hide the entire page (except the date picker)
+        // from screen readers to prevent document navigation
+        // (by headings, etc.) while the popup is open
+        this.$html.find('.container').attr('aria-hidden','true');
```

Comment:
A future version may need these containers to be configurable as it's very tightly bound to the Pulsar layout here